order/interfaces.go

```diff
@@ -256,6 +274,26 @@ var (
 	ZeroNonce Nonce
 )
 
+// MatchAnnouncementConstraints returns true when the asker announcement
+// constraints match the bidder announcement preferences.
+func MatchAnnouncementConstraints(asker ChannelAnnouncementConstraints,
```

Comment:
Nice idea to put it here, makes a lot of sense.
This could even be a method on the `ChannelAnnouncementConstraints` type if we want to be fancy.